### PR TITLE
QGCApplication: Use forward declaration in QQmlApplicationEngine

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
ping @DonLakeFlyer

File: src/QGCApplication.h [hunk 1]
```diff
@@ -30,6 +29,8 @@
 #include "OpalLink.h"
 #endif
 
+class QQmlApplicationEngine;
```
Comment (DonLakeFlyer) [inline src/QGCApplication.h:32]: Can you move this down with the other forward references and remove the obvious comment about circular header include.

Comment (patrickelectric) [reply src/QGCApplication.h:32]: This is not fixing any circular header includes, as the comment in the top of the others forward references. But Ok.

**Comment by patrickelectric** [comment]
Done.

**Pull request merged**